Framework/Utils/include/DPLUtils/RawParser.h

```diff
@@ -463,9 +475,11 @@ class RawParser
     }
 
     /// get header as specific type
+    /// Normal usage is get_if<T>() but in some rare cases the type can also be passed by parameter
+    /// get_if((T*)nullptr), the parameter is ignored
     /// @return pointer to header of the specified type, or nullptr if type does not match to actual type
     template <typename U>
-    U const* get_if() const
+    U const* get_if(U const* = nullptr) const
```

Comment:
@matthiasrichter As far as I understand, using this method requires compile time knowledge of the RDH version.
Cannot you provide more generic getter, in the same spirit as https://github.com/AliceO2Group/AliceO2/blob/8686319aa4c98344b64429198d41b54bafbf25e6/Framework/Utils/include/DPLUtils/RawParser.h#L266 ?

Reply:
yes, that's how it is, this has not changed in the PR. And no, the method you cite is creating the variant with one concrete alternative depending on the RDH version. If you want to get a concrete type out of that you have to specify it at compile time, same as if you do a `reinterpret_cast` on the raw pointer. Its the same like if you want to check for an alternative in std::variant. The different RDH versions are different types.

But you triggered an idea. We possibly can add a method `get_as` which does some conversion and default initialization of members not existing in one or the other. But `get_as<RDHv6>()` would give you an RDH v6 object also on RDH v5 data, but with default initialization for the new members.

So with `get_if` one can check for a concrete type and decide within the code on separate code branches, with `get_as` one can write code using commonalities of the RDH versions.

What do you think?

Reply:
I do more or less the same with a separate class https://github.com/AliceO2Group/AliceO2/blob/a9795f52172964719fe3f8d6ba80aa1bc94f8964/Detectors/Raw/include/DetectorsRaw/RDHUtils.h#L81 which can be morphed to any RDH version. 

To use the dplparser, I simply fetch reinterpret the raw pointer from the iterator as this placeholder RDHAny:
https://github.com/AliceO2Group/AliceO2/blob/a9795f52172964719fe3f8d6ba80aa1bc94f8964/Detectors/ITSMFT/common/reconstruction/src/RawPixelDecoder.cxx#L165

Seems to work w/o problems an w/o particular overheads.

Reply:
ok, if you move the RDHAny to a separate header file in DataFormats/Headers then I simply include this in the parser. see my comments in #3408 Do you have a benchmark assessing the overhead? Sure, can be optimized in the future.